lib/tx.js

```diff
@@ -74,10 +77,14 @@ function getPollFn(sdk, res, ref) {
     // Get the poll function
     var pollLink = util.getLink(res, 'next', 'poll');
     function pollFn() {
-      var href = pollLink.href;
+      var opts = {};
+      if (autoPush !== undefined && autoPush !== null) {
+        opts.autoPush = (typeof autoPush === 'function') ? autoPush() : !!autoPush;
```

Comment:
autoPush now takes a boolean / function. It sets the option on every poll to account for the scenario when the checkbox is clicked during an active poll

Reply:
Interesting. I wonder why this didn't come up for `rememberDevice`. It'd have the same problem, right?

Reply:
Does this need to be a function for the simple `verify` case, or is the `poll` case sufficient?

Reply:
Correct, I don't think the rememberDevice value will change if the checkbox is clicked during an active poll. However, we are implementing it this way for autoPush to keep it consistent with the way the autoPush cookie is set on the front end now.
It doesn't need to be a function for the verify case, I just thought it would be better to keep it consistent. What do you think?

Reply:
Good point Len, it is interesting cause it definitely has the same problem. I was just checking and when you click verify, we send ```rememberDevice=true```, but for polling we do not send that. And if we change the checkbox, then we don't change the value either. I do not know how the backend works for rememberDevice though... maybe we can have a follow up jira for that?
![pollingrememberdevice](https://user-images.githubusercontent.com/6979296/36181268-fd699896-10d8-11e8-921b-7aab98fa03fb.png)

and yes, for verify it doesn't need to be a function, but my vote goes for the consistency.

Reply:
re: verify - I'm down with consistency
re: `rememberDevice` - I'm curious if the server maintains the `rememberDevice` state for us. If it handles the `rememberDevice` state, it seems natural that it'd also handle the `autoPush` state. If it doesn't, we can do what this PR does (send it with every request). Is this a question for @srinivasanagandla-okta ?

Reply:
@kentonsmith-okta would you know the answer to the above ^ ?

Reply:
Spoke to @kentonsmith-okta about this. The backend seems to cache the rememberDevice value sent on the first verify and then save it on a successful authentication. If the user were to change the value during a poll, it would have no affect. The only way to clear a saved (true) value is to clear the browser's cookies.